### PR TITLE
os/bluestore: allow use BtreeAllocator 

### PR DESCRIPTION
**Pull request by NUABO**
BtreeAllocator was added in the commit
https://github.com/ceph/ceph/pull/41828.
Its performance has advantages in some scenarios,
but we cannot configure to use btree mode.

Fixes: https://tracker.ceph.com/issues/65678
Signed-off-by: tan changzhi <544463199@qq.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by ifed01 — APPROVED** [review]

**Review by ifed01 — CHANGES_REQUESTED** [review]
Please fix bluefs_allocator as well

File: src/common/options/global.yaml.in [hunk 1]
```diff
@@ -4143,6 +4143,7 @@ options:
   - bitmap
   - stupid
   - avl
+  - btree
```
Comment (NUABO) [inline src/common/options/global.yaml.in:4146]: hi @ifed01 , thank you for your reply, BtreeAllocator support for bluefs_allocator has been added here, are there any other problems? Looking forward to your reply.

**Comment by ifed01** [comment]
jenkins test make check

File: src/common/options/global.yaml.in (hunk 1 above)
Comment (ifed01) [reply src/common/options/global.yaml.in:4146]: Looks good to me now - will approve once all the basic checks below are done

Comment (NUABO) [reply src/common/options/global.yaml.in:4146]: That sounds pretty good

Comment (NUABO) [reply src/common/options/global.yaml.in:4146]: > Looks good to me now - will approve once all the basic checks below are done
That sounds pretty good

**Review by ifed01 — APPROVED** [review]

**Comment by tchaikov** [comment]
tested at https://pulpito.ceph.com/kchai-2024-05-04_14:52:58-rados-wip-kefu-testing-2024-05-04-distro-default-smithi/ . but due to the test failures. need to bisect the offending change or to see they are known issues.

**Comment by NUABO** [comment]
> tested at https://pulpito.ceph.com/kchai-2024-05-04_14:52:58-rados-wip-kefu-testing-2024-05-04-distro-default-smithi/ . but due to the test failures. need to bisect the offending change or to see they are known issues.

I checked the reason why the task is dead, and it should not be related to this modification~

**Comment by aclamk** [comment]
This PR passed rados suite.
https://pulpito.ceph.com/akupczyk-2024-07-12_08:52:40-rados-aclamk-testing-ganymede-2024-07-10-1144-distro-default-smithi/
https://pulpito.ceph.com/akupczyk-2024-07-11_11:24:14-rados-aclamk-testing-ganymede-2024-07-10-1144-distro-default-smithi/

**Pull request merged**